File: project2client/WebContent/controllers/friendcontroller.js
/**
 * 
 */
app.controller('FriendCtrl',function($scope,$location,FriendService){
	//STATEMENT
	function getAllSuggestedUsers(){
	FriendService.getAllSuggestedUsers().then(
			function(response){
			  $scope.suggestedUsers=response.data //Array of User objects
			},
			function(response){
				if(response.status==401)
					$location.path('/login')
			})
	}
	
	function getPendingRequests(){
		FriendService.getPendingRequests().then(
				function(response){
					$scope.pendingRequests=response.data //Array of Friend Object[{fromId,toId,status,friendId},{}]
				},
				
				function(response){
					if(response.status==401)
						$location.path('/login')
				})
	}
	
	getAllSuggestedUsers()
	getPendingRequests()
	$scope.addFriend=function(toId){//toId is suggestedUser in suggesteduserslist.html,toId is user object
		FriendService.addFriend(toId).then(
				function(response){
					getAllSuggestedUsers()//S in (A - (B U C))
				},
				function(response){
					if(response.status==401)
						$location.path('/login')
				})
	}
	
	
	$scope.acceptRequest=function(pendingRequest){//pendingRequest has fromId,toId,status,friendId
		FriendService.acceptRequest(pendingRequest).then(
				function(response){
					//get pending requests
					getPendingRequests()//from Friend where toId_email=? and status='P'
				},
				function(response){
					if(response.status==401)
						$location.path('/login')
				})
	}
	
	$scope.deleteRequest=function(pendingRequest){
		FriendService.deleteRequest(pendingRequest).then(
				function(response){
					//get pending requests
					getPendingRequests()//from Friend where toId_email=? and status='P'
				},
				function(response){
					if(response.status==401)
						$location.path('/login')
				})
	}
	
	function getAllFriends(){
    	FriendService.getAllFriends().then(
    			function(response){
    				//response.data is Array of User object
    				$scope.friends=response.data
    			},function(response){
    				if(response.status==401)
    					$location.path('/login')
    			})
    }
	
	getAllFriends()
})

File: project2middleware/src/main/java/com/niit/controllers/FriendController.java
package com.niit.controllers;

import java.util.List;

import javax.servlet.http.HttpSession;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

import com.niit.dao.FriendDao;
import com.niit.dao.UserDao;
import com.niit.models.ErrorClazz;
import com.niit.models.Friend;
import com.niit.models.User;

@RestController
public class FriendController {
	@Autowired
private FriendDao friendDao;
	@Autowired
private UserDao userDao;
	@RequestMapping(value="/suggestedusers",method=RequestMethod.GET)
	public ResponseEntity<?> getAllSuggestedUsers(HttpSession session){
		//Check for Authenticated- only logged user can post a blog 
				String email=(String)session.getAttribute("loginId");
				if(email==null){//if the user is not yet logged in,user is not an authenticated user
					ErrorClazz errorClazz=new ErrorClazz(5,"Please login..");
					return new ResponseEntity<ErrorClazz>(errorClazz,HttpStatus.UNAUTHORIZED);
				}
				List<User> suggestedUsers=friendDao.getAllSuggestedUsers(email);
				return new ResponseEntity<List<User>>(suggestedUsers,HttpStatus.OK);
	}
	@RequestMapping(value="/addfriend",method=RequestMethod.POST)
	public ResponseEntity<?> addFriend(HttpSession session,@RequestBody User toId){
		//Check for Authenticated- only logged user can post a blog 
		String email=(String)session.getAttribute("loginId");
		if(email==null){//if the user is not yet logged in,user is not an authenticated user
			ErrorClazz errorClazz=new ErrorClazz(5,"Please login..");
			return new ResponseEntity<ErrorClazz>(errorClazz,HttpStatus.UNAUTHORIZED);
		}
		User fromId=userDao.getUser(email);
		Friend friend=new Friend();
		friend.setFromId(fromId);
		friend.setToId(toId);
		friend.setStatus('P');
		
		friendDao.addFriend(friend);
		return new ResponseEntity<Void>(HttpStatus.OK);
	}
	@RequestMapping(value="/pendingrequests",method=RequestMethod.GET)
	public ResponseEntity<?> pendingRequests(HttpSession session){
		//Check for Authenticated- only logged user can post a blog 
				String email=(String)session.getAttribute("loginId");
				if(email==null){//if the user is not yet logged in,user is not an authenticated user
					ErrorClazz errorClazz=new ErrorClazz(5,"Please login..");
					return new ResponseEntity<ErrorClazz>(errorClazz,HttpStatus.UNAUTHORIZED);
				}
				List<Friend> pendingRequests=friendDao.getPendingRequests(email);
				return new ResponseEntity<List<Friend>>(pendingRequests,HttpStatus.OK);
	}
	@RequestMapping(value="/acceptrequest",method=RequestMethod.PUT)
	public ResponseEntity<?> acceptRequest(HttpSession session,@RequestBody Friend pendingRequest){
		String email=(String)session.getAttribute("loginId");
		if(email==null){//if the user is not yet logged in,user is not an authenticated user
			ErrorClazz errorClazz=new ErrorClazz(5,"Please login..");
			return new ResponseEntity<ErrorClazz>(errorClazz,HttpStatus.UNAUTHORIZED);
		}
		//update the status as 'A'
		pendingRequest.setStatus('A');
		friendDao.acceptRequest(pendingRequest);
		return new ResponseEntity<Void>(HttpStatus.OK);
	}
	@RequestMapping(value="/deleterequest",method=RequestMethod.PUT)
	public ResponseEntity<?> deleteRequest(HttpSession session,@RequestBody Friend pendingRequest){
		String email=(String)session.getAttribute("loginId");
		if(email==null){//if the user is not yet logged in,user is not an authenticated user
			ErrorClazz errorClazz=new ErrorClazz(5,"Please login..");
			return new ResponseEntity<ErrorClazz>(errorClazz,HttpStatus.UNAUTHORIZED);
		}
		friendDao.deleteRequest(pendingRequest);
		return new ResponseEntity<Void>(HttpStatus.OK);
	}
	
	@RequestMapping(value="/listoffriends",method=RequestMethod.GET)
	public ResponseEntity<?> listOfFriends(HttpSession session){
		String email=(String)session.getAttribute("loginId");
		//NOT LOGGED IN
		if(email==null){
			ErrorClazz errorClazz=new ErrorClazz(6,"Please login...");
    		return new ResponseEntity<ErrorClazz>(errorClazz,HttpStatus.UNAUTHORIZED);//login.html
		}
		List<User> friends=friendDao.listOfFriends(email);
		return new ResponseEntity<List<User>>(friends,HttpStatus.OK);
	}
}








